src/api.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+const assert = require('assert');
+const brorand = require('brorand');
+const hashjs = require('hash.js');
+const elliptic = require('elliptic');
+const Ed25519 = elliptic.eddsa('ed25519');
+const Secp256k1 = elliptic.ec('secp256k1');
+const addressCodec = require('ripple-address-codec');
+const deriveSecret = require('./secp256k1').deriveSecret;
+const utils = require('./utils');
+const hexToBytes = utils.hexToBytes;
+const bytesToHex = utils.bytesToHex;
+
+function generateSeed(options = {}) {
+  assert(!options.entropy || options.entropy.length >= 16, 'entropy too short');
+  const entropy = options.entropy ? options.entropy.slice(0, 16) : brorand(16);
+  return addressCodec.encodeSeed(entropy, options.algorithm || 'secp256k1');
+}
+
+const secp256k1 = {
+  deriveKeypair: function(entropy, options) {
+    const prefix = '00';
+    const privateKey = prefix + deriveSecret(entropy, options)
+      .toString(16, 64).toUpperCase();
+    const publicKey = bytesToHex(
```

Comment:
Not *really*. While I know I can get by with just eating rice, on the floor in a zen empty room, I like steak and mushroom sauce.

In general, I find it convenient to be able to import a one arg routine from a library rather than go through steps. For example, these are the steps I have to do to get a node public key in the format required by the rippled.cfg:

```js
const alias = 'N1';
const entropy = Sha512.quarter(alias);
const seed = generateSeed(entropy);
const keys = deriveKeypair(seed, {node: true});
const publicKey = encodeNodePublic(hexToBytes(keys.publicKey));
```

Note that I have to encode entropy into a seed string to pass to deriveKeypair.